### ISSUE TITLE
Adding a script to the docs?

### ISSUE THREAD

**Issue by capybaralet** [mila-udem/blocks#788]
I have a script that is meant as a quick-start guide to using blocks, included below.  I would like to add it to the docs if possible.

My personal experience with using Blocks has involved many false starts, because I could not easily figure out what is required vs. optional, and some of the terminology was confusing, etc.  I suspect that many other potential users are being turned off by the non-trivial barrier to entry.

Once I finally DID figure it out, I was surprised by how simple it was, and so I made this script, which is basically the first thing I personally would've liked to see in the docs.

I'm not really sure how to incorporate this into the online docs (another thing I found kind of annoying with these is that you have to copy/paste code chunks and remove the ">>>" (or maybe I just don't know what I'm doing), so I prefer to keep it as a script, personally).

Lemme know what you think.

https://github.com/capybaralet/Blocks_quickstart/blob/master/basic_blocks_script.py

"""
This script is a starting point for new Blocks users already familiar with
 Machine Learning and Theano.

We demonstrate how to use blocks to train a generic set of parameters (theano
 shared variables) that influence some arbitrary cost function (theano
 symbolic variable), so you can start using blocks features (e.g. monitoring,
 extensions, training algorithms) with your Theano code today.

To run an experiment, we simply construct a main_loop.MainLoop and call its
 run() method.  It suffices to pass the MainLoop a blocks.model.Model
 (which needs the cost), a blocks.algorithms.TrainingAlgorithm (which needs the
 cost and parameters), and a fuel.streams.DataStream*

As it is the script will run indefinitely, with no output.  You can interrupt
 training training anytime with Ctrl+C, or termination conditions can be added
 via extensions.

*The DataStream object is part of the partner library Fuel
 (https://github.com/mila-udem/fuel).
"""

import numpy
np = numpy
import theano
import theano.tensor as T

# (Here we make a toy dataset of two 2D gaussians with different means.)
num_examples = 1000
batch_size = 100
means = np.array([[-1., -1.], [1, 1]])
std = 0.5
labels = np.random.randint(size=num_examples, low=0, high=1)
features = means[labels, :] + std * np.random.normal(size=(num_examples, 2))
labels = labels.reshape((num_examples, 1)).astype(theano.config.floatX)
features = features.astype(theano.config.floatX)

# Define "data_stream"
from collections import OrderedDict
from fuel.datasets import IndexableDataset
# The names here (e.g. 'name1') need to match the names of the variables which
#  are the leaves of the computational graph for the cost.
dataset = IndexableDataset(
              OrderedDict([('name1', features), ('name2', labels)]))
from fuel.streams import DataStream, ForceFloatX
from fuel.schemes import SequentialScheme
data_stream = ForceFloatX(DataStream(dataset,
                                     iteration_scheme=SequentialScheme(
                                       dataset.num_examples, batch_size)))

# Define "cost" and "parameters"
# (We use logistic regression to classify points by distribution)
inputs = T.matrix('name1')
targets = T.matrix('name2')
ninp, nout = 2, 1
W = theano.shared(.01*np.random.uniform(
                        size=((ninp, nout))).astype(theano.config.floatX))
b = theano.shared(np.zeros(nout).astype(theano.config.floatX))
output = T.nnet.sigmoid(T.dot(inputs, W) + b)
# a theano symbolic expression
cost = T.mean(T.nnet.binary_crossentropy(output, targets))
# a list of theano.shared variables
parameters = [W, b]

# wrap everything in Blocks objects and run!
from blocks.model import Model
model = Model([cost])
from blocks.algorithms import GradientDescent, Scale
algorithm = GradientDescent(cost=cost,
                            parameters=parameters,
                            step_rule=Scale(learning_rate=.01))
from blocks.main_loop import MainLoop
my_loop = MainLoop(model=model,
                   data_stream=data_stream,
                   algorithm=algorithm)
my_loop.run()

**Comment by dwf** [comment]
I'm in favour of the general idea, but as is, the demo ignores pretty much the single most idiomatic thing about code written with Blocks, i.e. the use of bricks.

> I'm not really sure how to incorporate this into the online docs (another thing I found kind of annoying with these is that you have to copy/paste code chunks and remove the ">>>" (or maybe I just don't know what I'm doing), so I prefer to keep it as a script, personally).

The reason to include `>>>` and `...` is to a) visually demarcate them as examples, as if they were being run at an interpreter prompt, complete with output, b) allow the docs to be unit tested as [doctests](https://docs.python.org/3/library/doctest.html). This ensures that documentation stays in sync with the code as the code changes. Any decent editor (e.g. [example #1](http://vimdoc.sourceforge.net/htmldoc/visual.html#visual-block), [example #2](https://www.gnu.org/software/emacs/manual/html_node/emacs/Rectangles.html) should have support for rectangular selections (IPython also strips all that stuff from pastes by default).

**Comment by capybaralet** [comment]
Yeah, I was thinking of adding a little note saying that you can use bricks
to build the cost graph and pointing to the page where that's explained.

My goal here was to present the bare minimum of what is needed to start
using the library, although it's true you could also just use the blocks to
build graphs and use your own main loop, etc....

I guess that modularity should be emphasized more, somehow.


The >>>s read nice, and it is not just that, it is also just having the
code bits spread out instead of lumped together.  It's just so much faster
to start using it if you can just do one copy paste.




On Tue, Aug 4, 2015 at 9:11 PM, David Warde-Farley <notifications@github.com

**Comment by dwf** [comment]
Closing this as AFAIK there's been no pull request.
